airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -55,21 +55,20 @@ def url_base(self) -> str:
         return f"https://{self.shop}.myshopify.com/admin/api/{self.api_version}/"
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        decoded_response = response.json()
-        if len(decoded_response.get(self.data_field)) < self.limit:
-            return None
+        # Getting next page link
+        next_page = response.links.get("next", None)
+        if next_page:
+            next_page = next_page.get("url").split("page_info=")[1]
```

Comment:
this should use a query param parser to extract all params in the url rather than manually splitting